proxy/src/coco.rs

```diff
@@ -72,10 +74,10 @@ impl Peer {
     /// [`std::sync::Mutex::lock`] for further details.
     pub fn with_api<F, T>(&self, f: F) -> Result<T, error::Error>
     where
-        F: FnOnce(&net::peer::PeerApi) -> T,
```

Comment:
This caused more headaches than it solved, so I opted for returning `Result<T, error::Error>` and avoided `flatten` everywhere